### PR TITLE
Implement general http service

### PR DESCRIPTION
**Pull request by yarsanich**
First Implementations.

### CODE REVIEW

File: frontend/services/httpGeneral.service.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	'use strict';
+
+	angular
+		.module('base')
+		.service('httpGeneral',httpGeneral);
+
+	httpGeneral.$inject = [
+		'$http',
+	]
```
Comment (ihorhabshiy) [inline frontend/services/httpGeneral.service.js]: ;

File: frontend/services/httpGeneral.service.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+	function httpGeneral($http){
+		this.httpSend = function(object){
+			if (typeof object.url !== 'string' || object.url === undefined || object.url.length === 0){
+				console.log("HTTP REQUEST EROR: REQUEST ULR IS ABSENT");
```
Comment (ihorhabshiy) [inline frontend/services/httpGeneral.service.js]: лучше throw'нуть еррор, шоб красным был в консоли

File: frontend/services/httpGeneral.service.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+				return;
+			}
+			if (typeof object.url !== 'string' || object.type === undefined || object.type.length === 0){
```
Comment (ihorhabshiy) [inline frontend/services/httpGeneral.service.js]: `typeof object.url !== 'string'`

object.url or object.type ?

File: frontend/services/httpGeneral.service.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+				case 'put':
+					$http.put(object.url,object.body).then(succesfullRequest,failedRequest);
+					break;
+				case 'delete':
+					$http.delete(object.url).then(succesfullRequest,failedRequest);
+					break;
+				}
+			function succesfullRequest(res){
+				console.log("Succesfull Request");
+				alert(object.successMessageToUser);
```
Comment (ihorhabshiy) [inline frontend/services/httpGeneral.service.js]: Ну и вообще должен возвращать промис.

File: frontend/services/httpGeneral.service.js (hunk 1 above)
Comment (DmitrijDN) [inline frontend/services/httpGeneral.service.js]: httpGeneral.es6

We are writing using ecmasript2015

File: frontend/services/httpGeneral.service.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+	angular
+		.module('base')
+		.service('httpGeneral',httpGeneral);
+
+	httpGeneral.$inject = [
+		'$http',
+	]
+
+	function httpGeneral($http){
+		this.httpSend = function(object){
```
Comment (DmitrijDN) [inline frontend/services/httpGeneral.service.js]: object property description required

File: frontend/services/httpGeneral.service.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+					break;
+				case 'put':
+					$http.put(object.url,object.body).then(succesfullRequest,failedRequest);
+					break;
+				case 'delete':
+					$http.delete(object.url).then(succesfullRequest,failedRequest);
+					break;
+				}
+			function succesfullRequest(res){
+				console.log("Succesfull Request");
```
Comment (DmitrijDN) [inline frontend/services/httpGeneral.service.js]: for debug this console.log works fine, but when we start use this service our console will be overfilled with such messages

File: frontend/services/httpGeneral.service.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+				alert(object.successMessageToUser);
+			}	
+			function failedRequest(error){
+				if (object.errorMessageToDev) {
+        			if (error.data && typeof error.data && error.data.message)
```
Comment (DmitrijDN) [inline frontend/services/httpGeneral.service.js]: format code please. if should be in one tab from previous. Also, if you use if without {}, implementation should be on one string with if

File: frontend/services/httpGeneral.service.es6 [hunk 8]
```diff
@@ -33,25 +42,25 @@
...
 				//alert(object.successMessageToUser);
 				return res.data;
 			}	
 			function failedRequest(error){
 				if (object.errorMessageToDev) {
-        			if (error.data && typeof error.data && error.data.message)
-            			console.log(object.errorMessageToDev + ' ' + error.data.message);
-        			else 
-        				console.log(object.errorMessageToDev + ' ' + error.data);
+        			if (error.data && typeof error.data && error.data.message){
```
Comment (DmitrijDN) [inline frontend/services/httpGeneral.service.es6]: I guess if(error.data && error.data.message)  checking will work fine

**Pull request closed without merging**